doc/python/v6-changes.md

```diff
@@ -0,0 +1,157 @@
+---
+jupyter:
+  jupytext:
+    notebook_metadata_filter: all
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: '1.3'
+      jupytext_version: 1.16.4
+  kernelspec:
+    display_name: Python 3 (ipykernel)
+    language: python
+    name: python3
+  language_info:
+    codemirror_mode:
+      name: ipython
+      version: 3
+    file_extension: .py
+    mimetype: text/x-python
+    name: python
+    nbconvert_exporter: python
+    pygments_lexer: ipython3
+    version: 3.11.10
+  plotly:
+    description: Guide to changes in version 6 of Plotly.py and how to migrate from
+      version 5
+    display_as: file_settings
+    language: python
+    layout: base
+    name: Changes in Version 6
+    order: 8
+    page_type: example_index
+    permalink: python/v6-migration/
+    thumbnail: thumbnail/v4-migration.png
+---
+
+This page outlines the changes in Plotly.py version 6 and cases where you may need to update your charts or tools that you use for working with Plotly.py.
+
+<!-- #region -->
+## Jupyter Notebook Support
+
+Versions of Jupyter Notebook earlier than version 7 are no longer supported. To upgrade to the latest Jupyter Notebook:
+
+```
+pip install notebook --upgrade
+```
+
+## Change to anywidget for go.FigureWidget
+
+[go.FigureWidget](https://plotly.com/python/figurewidget/) now uses [anywidget](https://anywidget.dev/). Install `anywidget` with:
+
+```python
+pip install anywidget
+```
+
+## Processing NumPy and NumPy-Convertible Arrays
+
+Plotly.py now takes advantage of recent changes in how Plotly.js handles typed arrays for improved performance. See the [performance page](python/performance/) for more details.
+
+## Dataframe Support
+
+Plotly Express now uses [Narwhals](https://narwhals-dev.github.io/narwhals/) to natively support pandas, Polars, and PyArrow.
+
+## Removed Attributes
+
+The following attributes have been removed in Plotly.py 6.
+
+### `titlefont`,`titleposition`, `titleside`, and `titleoffset`
+
+The `titlefont`,`titleposition`, `titleside`, and `titleoffset` attributes have been removed. Replace them with `title.font`, `title.position`, `title.side`, and `title.offset`.
+
+Here's an example using `layout.titlefont`:
+
+```python
```

Comment:
@LiamConnors I don't love the idea of including code in the docs which specifically _won't_ work.

What if you combine the two code snippets into one, and include the old layout definition as commented-out code, followed by the new layout definition?

Also, maybe it would be clearer to define the layout using a `dict(title=dict(...))` rather than `go.Layout()` since the former is much more common in my experience — I'm not sure I've ever seen anyone use `go.Layout()`.

Reply:
Yeah, that's a fair point. The complete example probably isn't needed anyway. Reworked to add just a comment with the old method and to use dict rather than `go.Layout`